assets/src/hooks/DagreHook.ts

```diff
@@ -56,16 +74,42 @@ const DagreHook = {
         data.combos?.filter((combo) => !combo.parentId) || [];
       this.pushEvent("top-level-combos", topLevelCombos);
 
-      this.graph.data(data);
-
-      this.graph.render();
-
-      this.graph.changeSize(this.el.scrollWidth, this.el.scrollHeight);
+      const oldNodes = (this.graph.save() as GraphData).nodes || [];
+      const newNodes = data.nodes || [];
+
+      if (oldNodes.length === 0) {
+        this.graph.data(data);
+        renderGraph();
+        return;
+      }
+
+      const idsChanged = nodeIdsDifferent(oldNodes, newNodes);
+
+      if (idsChanged) {
+        this.graph.data(data);
+        if (hasInteracted) {
```

Comment:
I would rename it to something a little bit more descriptive, `hasInteractedSinceLastRefresh`, you can change `Refresh` to anything more meaningful I guess.

Reply:
Renamed in 3aa9e1c to `hasUserInteractedSinceLastRender`, couldn't find anything more concise 😄